.github/workflows/builder_docker-based_slsa3.yml

```
@@ -298,9 +299,11 @@ jobs:
           else
             username="${UNTRUSTED_SECRET_USERNAME:-${UNTRUSTED_INPUT_USERNAME}}"
             if [ "${username}" == "" ]; then
-                echo "registry-username is required." >&2
-                exit 1
+                # This may be the case for public base images.
+                echo "registry-username is not provided, skipping login." >&2
```

nit: You don't need to write this to stderr
```suggestion
                echo "registry-username is not provided, skipping login."
```